gen/denbi_portal_compute_center

```diff
@@ -140,48 +132,49 @@ sub convertToFriendlyName($) {
 
 ## creates structure for users.scim file
 sub processUsers {
-        my ($gid, $memberData) = @_;
+		my $groupId = shift;
+		my $memberId = shift;
+
+        my $userId = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_ID );
+		my $status = $data->getMemberAttributeValue( member => $memberId, attrName => $A_USER_STATUS );
 
-        my %memberAttributes = attributesToHash $memberData->getAttributes;
-        my $uid = $memberAttributes{$A_USER_ID};
-        my $status = $memberAttributes{$A_USER_STATUS};
 	
 	if($status ne $STATUS_VALID) { return; }
 
-	unless($userStruc->{$uid}) {
-		%memberAttributes = map { convertToFriendlyName($_) => $memberAttributes{$_} } keys %memberAttributes;
-		$userStruc->{$uid} = \%memberAttributes;
+	unless($userStruc->{$userId}) {
+		$userStruc->{$userId}->{"id"}=$userId;
+		$userStruc->{$userId}->{"status"}=$status;
+		$userStruc->{$userId}->{"preferredMail"}=$data->getUserAttributeValue( member => $memberId, attrName => $A_USER_EMAIL );
+		$userStruc->{$userId}->{"login-namespace:elixir-persistent"}=$data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LOGIN );
 	}
 
-        processMemberships $gid, $uid;
+        processMemberships $groupId, $userId;
 }
 
 ## creates structure for groups.scim file
 sub processGroups {
-        my $group = shift;
-        my %groupAttributes = attributesToHash $group->getAttributes;
-        my $membersElement = ($group->getChildElements)[1];
-
-	my $gid = $groupAttributes{$A_GROUP_ID};
-
-	#convert attributes data
-	$groupAttributes{$A_GROUP_DENBI_PROJECT_MEMBERS} = decode_json $groupAttributes{$A_GROUP_DENBI_PROJECT_MEMBERS};
-
-	unless(exists $groupStruc->{$gid}) {
-		%groupAttributes = map { convertToFriendlyName($_) => $groupAttributes{$_} } keys %groupAttributes;
```

Comment:
We have to check if the service used only defined attributes, which are defined at the beginning of the script. It seems that it could have generated also other attributes (commented at the beginning) as it iterated through all of the attributes on this line, not only the defined ones. @dBucik could you help us with that?

Reply:
Fixed, used methods from https://github.com/CESNET/perun/pull/3264